shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala

```diff
@@ -55,7 +55,6 @@ class PickleReader {
     val tOpt = entries(i).asInstanceOf[T | Null]
     if tOpt == null then {
       val res = pkl.unsafeFork(index(i))(op)
-      assert(entries(i) == null, entries(i))
```

Comment:
It would still be good to check that we don't overwrite the entry with something *different*. Perhaps
```scala
assert(entries(i) == null || (entries(i) eq res), entries(i))
```

Reply:
`entries(i)` and `res` are equal but not reference equal

Reply:
Hum, OK, I guess we have to take it as is then. It's a bit of a shame, though.